api/client/webclient/webclient.go

```diff
@@ -38,13 +38,22 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+type Config struct {
+	Context       context.Context
+	ProxyAddr     string
+	Insecure      bool
+	Pool          *x509.CertPool
+	ConnectorName string
+	ExtraHeaders  map[string]string
+}
+
 // newWebClient creates a new client to the HTTPS web proxy.
-func newWebClient(insecure bool, pool *x509.CertPool) *http.Client {
+func newWebClient(cfg *Config) *http.Client {
```

Comment:
Following up on my suggestion above:
```suggestion
func newWebClient(cfg *Config) (*http.Client, error) {
  if err := cfg.CheckAndSetDefaults(); err != nil {
    return nil, trace.Wrap(err)
  }
```